scripts/suite_objects.py

```diff
@@ -1184,6 +1194,41 @@ def analyze(self, phase, group, scheme_library, suite_vars, level):
                                                   vstdname))
                 # end if
             # end if
+            # Are there any forward/reverse transforms for this variable?
+            if compat_obj is not None and (compat_obj.has_vert_transforms or compat_obj.has_unit_transforms):
+                # Add local variable (<var>_local) needed for transformation.
+                tmp_var = var.clone(var.get_prop_value('local_name')+'_local')
+                alloc_stmt = "allocate({}({}))"
+                self.__alloc_transforms.append(alloc_stmt.format(var.get_prop_value('local_name')+'_local',''))
+                self.__group.manage_variable(tmp_var)
+
+                # Add local variable (<var>_nlay) needed for vertical flipping.
+                indices = [':']*var.get_rank()
+                if compat_obj.has_vert_transforms:
+                    verti_var = Var({'local_name':var.get_prop_value('local_name')+'_nlay',
+                                     'standard_name':var.get_prop_value('local_name')+'_nlay',
+                                     'type':'integer', 'units':'count', 
+                                     'dimensions':'()'}, _API_LOCAL, self.run_env)
+                    self.__group.manage_variable(verti_var)
+                    # Set indices for vertical flipping.
+                    dim = find_vertical_dimension(var.get_dimensions())
+                    for dpart in dim[0].split(':'):
+                        indices[dim[1]] = var.get_prop_value('local_name')+'_nlay:1:-1'
+                    # Create statement for use in write stage.
+                    write_stmt = var.get_prop_value('local_name')+"_nlay = size({},{})"
```

Comment:
I don't think this is valid. I do not think there is any requirement that the array size has to match the dimension. With Fortran 90-style array declarations (as in the var action test), you should be able to get correct behavior even if they do not match. Also, I do not see why you need a separate local variable. Why just not use `nlev`?